File: numeric/max/max.rb

def max_list(list)
    current = 0
    list.each do |value|
        if value > current
            current = value
        end
    end
    return current
end

list = [15, 10]
if ARGV.length > 0
    list = ARGV.map { |arg| arg.to_i }
end

maxValue = max_list(list)

puts "values: #{list}"
puts "max: #{maxValue}"

File: numeric/max/max.kt
fun max(values: List<Int>): Int {
    var current = 0;
    for (test in values) {
        if (test > current) {
            current = test;
        }
    }
    return current;
}

fun main(args: Array<String>) {
    var list = listOf(15, 10)
    if (args.size > 0) {
        list = args.map { arg -> arg.toInt() }
    }

    val maxValue = max(list)

    println("values: $list")
    println("max: $maxValue")
}
